### DISCUSSION TITLE
Is the file transfer speed limit only 20Mbps?

### DISCUSSION THREAD

**Discussion by xuexiaokang** [rustdesk/rustdesk#1607] [Q&A] [answered]
Is the file transfer speed limit only 20Mbps? My uplink bandwidth is 40Mbps, but the file transfer is always only 20Mbps, can I remove this limit in RustDesk?

**Comment by Angell1993** [comment]
Why is my file transfer limited to 2MB/s and after 5min it goes down to 500KB/s??

I use my own rustdesk server setup, on a 1gbit/1gbit ethernet connection.. Is there some settings that can speed this up, seeing I'm already using my own server??

**Comment by xlionjuan** [comment, marked as answer]
Yes, it is possible, just set this args to your hbbr, I just copied my args from my docker-compose.yml

```
     - "SINGLE_BANDWIDTH=xxx" # It is Mbps
```

https://github.com/rustdesk/rustdesk-server?tab=readme-ov-file#env-variables
https://github.com/rustdesk/rustdesk-server/discussions/371

**Reply by BA-24** [reply]
Petition to change this default? 2mbps is terribly slow... The entire point of self-hosting is to have it be fast, right?

**Reply by xlionjuan** [reply]
You could just modify it though env variable

**Reply by Jan-Yan** [reply]
This method is very effective, thank you. @xlionjuan

**Reply by miro-igov** [reply]
What is the exact procedure of increasing the SINGLE_BANDWIDTH? 
I added bw.env file in /var/lib/rustdesk/ with content SINGLE_BANDWIDTH=500 also added export SINGLE_BANDWIDTH=500 in ~/.bashrc but log always shows 
`INFO [src/relay_server.rs:146] SINGLE_BANDWIDTH: 16Mb/s` when starting the hbbr.

**Reply by xlionjuan** [reply]
Don't use install script

**Reply by miro-igov** [reply]
I do not understand what do you mean. Rustdesk is already installed, configured and used daily. I just need to increase this very low bandwidth limit. How?

**Reply by xlionjuan** [reply]
**Do not** use install scripts to install RustDesk Server, hbbs, hbbr, it is too complicated to maintain and adjust settings.

**Comment by mikekgr** [comment]
Please help me to understand, if I create a .env file at the same directory where the hbbr binary is, and then if the content of the .env file is:
SINGLE_BANDWIDTH=300

then am I ok in order to increase this value?|
If not please guide me.

Thanks

**Reply by xlionjuan** [reply]
Yes if your server's internet speed is enough

**Reply by mikekgr** [reply]
Thanks a lot !

**Comment by naddel91** [comment]
how can I change the max bandwidth in docker? I use rustdesk as a docker in portainer and 2 MB/s is way to slow for file transfers.

**Reply by xlionjuan** [reply]
Add `SINGLE_BANDWIDTH` env to your hbbr

Set it to you desired, it is Mbps, default is 16

(16/8=2MB/s)


https://github.com/rustdesk/rustdesk-server

**Reply by xlionjuan** [reply]
I don't know how you deployed though Portainer, `Stack`(compose) or `Add Container`?

No matter which, you could edit with `Duplicate/Edit` button after you click with your hbbr container, scroll to the bottom, find `Env`, add the env I mentioned above.

If you created with `Stack`(compose), it is recommend to edit your `Stack` with proper env and redeploy.

Ex:
```yaml
    environment:
      - "SINGLE_BANDWIDTH=900" #Mbps
```

**Reply by xlionjuan** [reply]
https://www.reddit.com/r/rustdesk/comments/1fv9z4h/change_server_settings_after_server_is_up/

**Reply by naddel91** [reply]
![rustdesk env](https://github.com/user-attachments/assets/40fbd327-9a16-4e4f-8633-9e6f2b63d12a)

here? it won't let me.

Update: Had to "re-deploy". Works like a charm. Thanks so much. Why does rustdesk have a limit there to begin with?

**Reply by xlionjuan** [reply]
Prevent you DDoS yourself just because someone doing file transfers, upload speed isn't large for most people, I increased it to 900Mbps because I deployed it in Linode.

**Reply by BA-24** [reply]
> Prevent you DDoS yourself just because someone doing file transfers, upload speed isn't large for most people, I increased it to 900Mbps because I deployed it in Linode.

The absolute vast majority of people nowadays have a network bandwidth higher than 20mbps ([source](https://www.speedtest.net/global-index)) and that's not accounting for the fact that people using this program are probably higher up on that spectrum as well so I find it very unlikely that anyone would "DDoS" themselves, and even if they do, they could adjust it down to a level that works well for them.

**Reply by xlionjuan** [reply]
If you have only 50Mbps upload, and you have other services other than RustDesk Server, you don't want RustDesk Server uses 100% upload resources and interfere your other services.

**Reply by BA-24** [reply]
First of all, 50mbps is more than double that of 20 and would be a way more reasonable limit IMO, besides that, I'm fairly certain all modern OS's/routers deal with this issue (I for one have never had any issues when maxing out my upload/download speeds which I frequently do), and even if they don't, you can still pause the upload/download or just lower the limit if you know you have slow internet

**Reply by xlionjuan** [reply]
The delay of the connection will increased if you run out of the upload resources, and most of home router's QoS is bad, and not enabled by default.

**Comment by naddel91** [comment]
DDoS yourself?!
I use it to transfer files to my server once in a while.

**Reply by xlionjuan** [reply]
Not everyone are build server for just for himself.

**Comment by mlopiccolo-progres** [comment]
I've used install.sh, how do I apply this environment variable?

**Reply by xlionjuan** [reply]
Systemd unit.

**Reply by mlopiccolo-progres** [reply]
Through the "EnvironmentFile" directive, right?

**Reply by xlionjuan** [reply]
If you have only one or few envs to set, there is no need to create separate env file.

**Reply by mlopiccolo-progres** [reply]
Okay, thank you very much!

**Comment by DataMeister1** [comment]
Could this sort of thing be made smarter so it knows if the connection is over the internet or using a local IP connection. My internet is fairly slow so the cap is fine for an internet connection, but I also have a few connections that use the LAN IP number to connect and those are slow too.

In fact for internet connections it might be nice if the speed could be set for individual clients. If I have one client on a 1 GB internet service and one on a 30 Mbps service, they should be able to handshake the details and come up with the best speed. So maybe the custom setting should have both a maximum internet speed and then a "use this percentage of the bandwidth" setting.

**Reply by xlionjuan** [reply]
The speed of direct connections WILL NOT be affected by the settings of hbbr.

**Reply by DataMeister1** [reply]
Oh. Well in that case I wonder why I have two PCs that are connecting locally through a gigabit switch but only transferring at 70 Mbps.
